src/renderer/views/Watch/Watch.js

```diff
@@ -1384,10 +1409,72 @@ export default defineComponent({
       ]
     },
 
-    createInvidiousDashManifest: function () {
+    createInvidiousDashManifest: async function () {
       let url = `${this.currentInvidiousInstance}/api/manifest/dash/id/${this.videoId}`
 
-      if (!process.env.IS_ELECTRON || this.proxyVideos) {
+      // If we are in Electron,
+      // we can use YouTube.js' DASH manifest generator to generate the manifest.
+      // Using YouTube.js' gives us support for multiple audio tracks (currently not supported by Invidious)
+      if (process.env.IS_ELECTRON) {
+        // Invidious' API response doesn't include the height and width (and fps and qualityLabel for AV1) of video streams
+        // so we need to extract them from Invidious' manifest
+        const response = await fetch(url)
+        const originalText = await response.text()
+
+        const parsedManifest = new DOMParser().parseFromString(originalText, 'application/xml')
+
+        /** @type {import('youtubei.js').Misc.Format[]} */
+        const formats = []
+
+        /** @type {import('youtubei.js').Misc.Format[]} */
+        const audioFormats = []
+
+        let hasMultipleAudioTracks = false
+
+        for (const format of this.adaptiveFormats) {
+          if (format.type.startsWith('video/')) {
+            const representation = parsedManifest.querySelector(`Representation[id="${format.itag}"][bandwidth="${format.bitrate}"]`)
+
+            format.height = parseInt(representation.getAttribute('height'))
+            format.width = parseInt(representation.getAttribute('width'))
+            format.fps = parseInt(representation.getAttribute('frameRate'))
+
+            // the quality label is missing for AV1 formats
+            if (!format.qualityLabel) {
+              format.qualityLabel = format.width > format.height ? `${format.height}p` : `${format.width}p`
+            }
+          }
+
+          const localFormat = convertInvidiousToLocalFormat(format)
+
+          if (localFormat.has_audio) {
+            audioFormats.push(localFormat)
+
+            if (localFormat.is_dubbed || localFormat.is_descriptive) {
+              hasMultipleAudioTracks = true
+            }
+          }
+
+          formats.push(localFormat)
+        }
+
+        if (hasMultipleAudioTracks) {
+          // match YouTube's local API response with English
+          const languageNames = new Intl.DisplayNames('en-US', { type: 'language' })
```

Comment:
Should we use the user's current locale instead of using "en-US"?

Reply:
My goal with this pull request was to match the local API, so that you would have the same user experience with both the local and the Invidious API.

If you want me to switch to using the users locale, I can do that, it'll just be a different user experience than with the local API or the Invidious API when it gets multiple audio track support.